### PR TITLE
Added type annotations to generate_deployments_for_service

### PR DESCRIPTION
**Pull request by solarkennedy**

### CODE REVIEW

File: paasta_tools/generate_deployments_for_service.py [hunk 1]
```diff
@@ -83,13 +125,13 @@ def get_cluster_instance_map_for_service(soa_dir, service, deploy_group=None, ty
...
         instances = get_instance_configs_for_service(soa_dir=soa_dir, service=service, type_filter=type_filter)
-    cluster_map = defaultdict(lambda: defaultdict(list))
+    cluster_map: Dict[str, Dict[str, List[str]]] = defaultdict(lambda: defaultdict(list))
     for instance_config in instances:
         cluster_map[instance_config.get_cluster()]['instances'].append(instance_config.get_instance())
     return cluster_map
 
 
-def get_latest_deployment_tag(refs, deploy_group):
+def get_latest_deployment_tag(refs, deploy_group: str) -> Tuple[str, str]:
```
Comment (EvanKrall) [inline paasta_tools/generate_deployments_for_service.py]: probably want to annotate `refs` too.

**Comment by solarkennedy** [comment]
Something about this code makes a "diff" in deployments.json, which is worrisome.

When I do the diff the 'deployments' section, it looks like deployments/controls are ordered differently. The total content looks the same though.

**Comment by EvanKrall** [comment]
What do you mean a "diff"? Like tests are failing equality assertions? Or literally the `diff` command says they're different.

**Comment by solarkennedy** [comment]
Literally diff. I did yelp-main to capture the largest set of differences. Disregard though, I ran it today and it had no differences.

**Pull request merged**